### PR TITLE
DOCS-#3610: Add section about handling of `read_csv` `skiprows` parameter

### LINKED ISSUE #3610: Add documentation describing handling of `skiprows` parameter by `read_csv` function in Modin

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3610  
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/flow/modin/engines/base/io.rst [hunk 1]
```diff
@@ -97,3 +97,96 @@ classes for reading files of different formats.
...
+  8
+  """
+
+  # `header` parameter absence is equivalent to `header="infer"` or `header=0`
+  # rows 1, 5, 6, 7, 8 are read with header "0"
+  df = pandas.read_csv(StringIO(data), skiprows=[2,3,4])
+  # rows 5, 6, 7, 8 are read with header "1", row 0 is skipped additionally
+  df = pandas.read_csv(StringIO(data), skiprows=[2,3,4], header=1)
+  # rows 6, 7, 8 are read with header "5", rows 0, 1 are skipped additionally
+  df = pandas.read_csv(StringIO(data), skiprows=[2,3,4], header=2)
```
Comment (YarShev) [inline docs/flow/modin/engines/base/io.rst:136]:
```suggestion
  df = pandas.read_csv(StringIO(data), skiprows=[2, 3, 4])
  # rows 5, 6, 7, 8 are read with header "1", row 0 is skipped additionally
  df = pandas.read_csv(StringIO(data), skiprows=[2, 3, 4], header=1)
  # rows 6, 7, 8 are read with header "5", rows 0, 1 are skipped additionally
  df = pandas.read_csv(StringIO(data), skiprows=[2, 3, 4], header=2)
```

File: docs/flow/modin/engines/base/io.rst [hunk 2]
```diff
@@ -97,3 +97,96 @@ classes for reading files of different formats.
...
+but the same logic is applicable for other types of the parameters.
+
+Processing `skiprows` by Modin
+==============================
+
+As it can be seen skipping rows in the pandas import functions is complicated and distributing
+this logic across multiple workers can complicate it even more. Thus in some rare corner cases
+default pandas implementation is used in Modin to avoid excessive Modin code complication.
+
+Modin uses two mechnisms of rows skipping:
```
Comment (YarShev) [inline docs/flow/modin/engines/base/io.rst:155]:
```suggestion
Modin uses two techniques for skipping rows:
```

**Comment by YarShev** [comment]
@amyskov, can #3511, #3419 and #3391 affect the handling of `skiprows` parameter documented here?

**Review by devin-petersohn — COMMENTED** [review]
@amyskov Should we try to get this in before release 0.12.0?

**Comment by amyskov** [comment]
> @amyskov, can #3511, #3419 and #3391 affect the handling of `skiprows` parameter documented here?

It shouldn't, this PR mostly describes `skiprows` processing logic that should comply with mentioned PRs changes.

**Comment by amyskov** [comment]
> @amyskov Should we try to get this in before release 0.12.0?

Not sure, i think we can put it to 0.12.1 release.

**Comment by vnlitvinov** [comment]
@amyskov what makes this a draft?

**Comment by amyskov** [comment]
> @amyskov what makes this a draft?

@vnlitvinov just checked that CI passed, marked as ready for review now.

**Review by vnlitvinov — COMMENTED** [review]
Mostly good, I've left some small proofreading and formatting fixes.

**Comment by amyskov** [comment]
> Mostly good, I've left some small proofreading and formatting fixes.

@vnlitvinov PR is ready for review.

**Review by YarShev — APPROVED** [review]

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Pull request merged**